extensions/eclipse-che-theia-remote-impl-che-server/src/node/che-server-remote-api-impl.ts

```diff
@@ -52,6 +51,6 @@ export class CheServerRemoteApiImpl {
   }
 
   getCheApiURI(): string {
-    return this.baseAPIUrl;
+    return this.baseAPIUrl!;
```

Comment:
Why forcing non-null assertion here?
Let's declare it in the return types - `getCheApiURI(): string | undefined {`

Reply:
I think `process.env.CHE_API_EXTERNAL` never should be null. So from api retrieving logic we always return value and undefined it something not expected. Maybe instead of console.error we should throw unexpected exception... And check that this error should be displayed on the che-theia ui. What do you think? Otherwise if we will change return type from `string` to `string | undefined` we have to apply undefined check in every usage place. And it will be not fair: actually code will be not working without che-api url. Container plugin, task plugin, factory plugin and so on can't work properly without interaction with che-server or dev controller analog.

Reply:
Actually, there's only one client of this method that already checks it and throws :slightly_smiling_face: 
https://github.com/eclipse-che/che-theia/blob/6fe4d5611400ec7a91ca959c692394d27d0091f0/extensions/eclipse-che-theia-remote-impl-che-server/src/node/che-server-factory-service-impl.ts#L22-L26
So, we can throw a bit earlier, in this constructor. But I'd say it's not mandatory for this PR.